### ISSUE TITLE
Tested with Claymore 10.0?

### ISSUE THREAD

**Issue by mhousser** [JuicyPasta/Claymore-No-Fee-Proxy#29]
Title says it all.

**Comment by mhousser** [comment]
That's too bad. Looking for very simple instructions on achieving this in Win 10 x64 + Claymore v10.0. This tool, or any other methods. Difficult to find clear step-by-step instructions.

**Comment by lcheung90** [comment]
It seems that in v10.0 Claymore does not use the specified pool and falls back to using us1.ethpool.org:3333 and bypasses the request_handler.

@JuicyPasta @drdada 
![claymorev10](https://user-images.githubusercontent.com/1421411/31592702-0d431f16-b1df-11e7-9d6a-c0fa037206e3.PNG)

172.12.10.228 is my fake WAN

**Comment by mhousser** [comment]
Do you know of any manual workaround? A Windows software tool that can
intercept and modify outgoing packets?

**Comment by lcheung90** [comment]
[Wireshark](https://www.wireshark.org/), but my experience with it is limited 

I found another proxy for Claymore, check it out [https://github.com/LostSoulFly/MinerProxy](url)
I have not tried it yet.

**Comment by mhousser** [comment]
Have looked at Wireshark, but it's so complex - found the proper packets for Claymore going out, but would really love to see a full step-by-step documentation on how to set it up to intercept/modify data.

I'll take a look at the link you just shared.

There must be some tutorial somewhere on the internet, I keep searching..

**Comment by drdada** [comment]
Hello, i have not tested CM v10.0 yet. #25 
And i don't think i will update/support it. (Lack of time and no longer motivations)
Feel free to read the code and create pull request. :+1: 

@mhousser Wireshark can only read packets (no forge or manipulations). I'm sorry if my guide are too succinct for you. If you want to manipulate some packet take a look at nfqueue (netfilter on linux).

@lcheung90 What are your args ? did you set something like _allcoin_ or _allpool_ ?

**Comment by lcheung90** [comment]
@drdada I had allpool set (I changed the WAN IP from before)

New script looks like this:
`ethdcrminer64.exe -epool 175.5.10.15:8008 -ewal 0x1a40fb8ac7a9801c652aab300eec3393d057e1eb.Miner01 -epsw x -mode 1 -tt 68
`

**Comment by lcheung90** [comment]
@drdada @mhousser 

OK, after removing the -allpool option it seems to work. 

However, the time to transmit the share to the pool seems to increase little by little and eventually you end up with quite a few stale shares. I am not sure if it is detecting the proxy, or the slow down is due to the Byzantium fork (faster block time.)

![stats](https://user-images.githubusercontent.com/1421411/31697522-006697fc-b36d-11e7-99bf-26abd8d869ee.PNG)

**Comment by lcheung90** [comment]
Going to run CM v10 without the proxy for a bit and see if the transmission times are the same... for science.

**Comment by mhousser** [comment]
I do enjoy a good science! 

Can you confirm in your screenshot that you now have 2 workers listed in ethermine.org pool..?

**Comment by lcheung90** [comment]
I only have one miner set up, but the additional one comes from the proxy (it creates a new worker <wallet>.rekt) I enabled the email notifications for "down" workers and got 4 emails roughly at the time the "rekt" worker went down. This happens because the "rekt" worker is not mining constantly.

**Comment by mhousser** [comment]
That's what I thought - the DevFee re-directed submission creates a 2nd worker in the pool. So you are seeing it? Nice.

I suppose your 2nd (fake) worker "rekt" is only mining 1% of the time, so you will get a lot of down notifications for it!

**Comment by lcheung90** [comment]
Nope, I got only 4 email notifications. 

Correct me if I am wrong, but I assume Claymore will not find a share every time the devfee process takes place and it is more luck than anything.

In a mining farm scenario, I can see a ton of notifications if the setting is enabled. So far, it seems like transmission times are faster without the proxy... let's see what happens in a couple of hours.

**Comment by lcheung90** [comment]
Alright, so using the proxy with CM v10 after Byzantium adds a to the transmission time. It can go as high as ~800ms. This translates to roughly 6-9% stale shares per hour for me versus 1-4% without using the proxy. 

It is also worth mentioning that after switching the proxy off Claymore crashed several times, don't know the reason.

So, it works with v10. Is it worth? You decide...

**Comment by mhousser** [comment]
Good work on your research! 

Hmm. It looks like maybe a packet modifying solution would be the only real way forward. 

There are solid linux examples of how to do this (full command line instructions on how to set it up), but can't find anything for Windows. A few people say they've set it up, but wouldn't share the details.

**Comment by lcheung90** [comment]
Well it is always easier to hack stuff up in Linux :) Have you tried those?

**Comment by mhousser** [comment]
No, we're definitely all Windows people, it would be a serious drawback if we had to learn the linux stack, including mining tools, AMD blockchain drivers, etc. But who knows. Maybe next rig we could experiment with that "EthOS" OS that seems to be popular. It's just *really* unappealing to go to Linux for us Windows-for-life people. =P

https://omnipacket.com/safepcap.html looks promising. Probably if you can build the right "filter expression" and do some clever regex, you could do it this way.

Then there should be virtually no effect on latency (in my mind, at least..)

**Comment by 53nsk** [comment]
You can try this: https://bitcointalk.org/index.php?topic=2274616.0

**Comment by lcheung90** [comment]
@mhousser  Are how many miners does your group have? :)

Just found this project... looks interesting. [Divert](https://github.com/basil00/Divert/wiki/WinDivert-Documentation#divert_recv_ex)

**Comment by febriyan1302** [comment]
Hi, i've got error on Claymore 10.
How to solve this problem ?
![screen shot 2017-10-26 at 1 38 23 pm](https://user-images.githubusercontent.com/18423871/32038234-26cf0a90-ba53-11e7-9065-c2250284cfd6.png)

**Comment by vstene** [comment]
Works fine on Ubuntu.

**Comment by radius1980** [comment]
i am sory, this is work after 4 hours, seen rekt. thnks
